src/utils/curve/curve.js

```diff
@@ -0,0 +1,30 @@
+import { retrieveData } from '../network'
+
+/**
+ * @function retrieve
+ * @memberof module:api.Curve
+ * @static
+ * @param {object} params - object of parameters:
+ * @param {number} params.AMId - Owning Asset Manager ID of the Profit and Loss record
+ * @param {Function} [callback] - Called with two arguments (error, result) on completion
+ * @returns {Promise|null} If no callback supplied, returns a Promise
+ */
+export function retrieve({ AMId, businessDate, assetIds }, callback) {
+  const params = {
+    AMaaSClass: 'curve',
+    AMId,
+    businessDate,
+    assetIds
```

Comment:
You can't pass these wholesale into `retrieveData`. This is the place to build `resourceId` from what is passed in e.g.
```javascript
export function retrieve({ AMId, businessDate, assetIds }, cb) {
  const params = {
    AMaaSClass: 'curve',
    AMId,
    resourceId: `${businessDate}/${assetIds}`
  }
  ...
}
```